src/main/java/frc/robot/VisionProcessor.java

```diff
@@ -132,19 +145,34 @@ public void process(Mat sourceFrame) {
         ArrayList<RotatedRect> initial = minimumBoundingRectangle(parent.filterContoursOutput());
         ArrayList<RotatedRect> findContours = minimumBoundingRectangle(parent.findContoursOutput());
         ArrayList<RotatedRect> ok = new GibberishRectangleFilter().filter(initial);
-        ArrayList<RotatedRect> selected = new DumbAndAloneRectangleFilter().filter(ok);
-
+        numRects = ok.size();
+
+        //reporter.getTargetAlignButtonPressed() == false || prevNumRects != numRects || frames.getTicks()<10 
+        //is the target lock logic, if all of these are false than target lock is initiated
+        if(reporter.getTargetAlignButtonPressed() == false || prevNumRects != numRects || frames.getTicks()<10){
```

Comment:
make a different class to keep track of targetAlign on/off.  maybe a class like TargetAlignControl

in there, you should check both the targetLockChecker, ( liek you are in getTargetAlignButtonPressed), AND the network table you're going to eventually set from the robot

Reply:
Additionally, this logic is at a lower level of abstraction than the above code, which is single-line-does-a-filter kind of code.